// skypicker/build/constants.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var SKYPICKER_BASE_API_URL = 'https://api.skypicker.com';

var LOCATION_TYPES = Object.freeze({
  AIRPORT: 'AIRPORT',
  AUTONOMOUS_TERRITORY: 'AUTONOMOUS_TERRITORY',
  CITY: 'CITY',
  COUNTRY: 'COUNTRY',
  STATION: 'STATION',
  SUBDIVISION: 'SUBDIVISION'
});

var LOCATION_RESULTS_SORT_TYPES = Object.freeze({
  ASCENDING_NAME: 'ASCENDING_NAME',
  ASCENDING_RANK: 'ASCENDING_RANK',
  DESCENDING_NAME: 'DESCENDING_NAME',
  DESCENDING_RANK: 'DESCENDING_RANK'
});

var FLIGHT_RESULTS_SORT_TYPES = Object.freeze({
  DATE: 'DATE',
  DURATION: 'DURATION',
  PRICE: 'PRICE',
  QUALITY: 'QUALITY'
});

var AIRLINES_FILTER_TYPE = Object.freeze({
  EXCLUDE: 'EXCLUDE',
  INCLUDE: 'INCLUDE'
});

exports.SKYPICKER_BASE_API_URL = SKYPICKER_BASE_API_URL;
exports.LOCATION_TYPES = LOCATION_TYPES;
exports.LOCATION_RESULTS_SORT_TYPES = LOCATION_RESULTS_SORT_TYPES;
exports.FLIGHT_RESULTS_SORT_TYPES = FLIGHT_RESULTS_SORT_TYPES;
exports.AIRLINES_FILTER_TYPE = AIRLINES_FILTER_TYPE;
